app/main.py

```diff
@@ -1,2 +1,159 @@
+from fractions import Fraction
+from typing import Any, Hashable, Iterable, Optional
+
+
 class Dictionary:
-    pass
+    class __Node:
+        def __init__(
+                self,
+                key: Optional[Hashable] = None,
+                value: Optional[Any] = None,
+                key_hash: Optional[int] = None
+        ) -> None:
+            self.key = key
+            self.value = value
+            self.key_hash = key_hash
+            self.deleted = False
+
+    def __init__(
+            self,
+            initial_data: Optional[Iterable[tuple[Hashable, Any]]] = None,
+            **kwargs
+    ) -> None:
+        self.__capacity = 8
+        self.__size = 0
+        self.__threshold = int(Fraction(2, 3) * self.__capacity)
+        self.__table = [Dictionary.__Node()] * self.__capacity
+
+        if initial_data:
+            self.update(initial_data)
+        if kwargs:
+            self.update(**kwargs)
+
+    def __repr__(self) -> str:
+        items = ", ".join(f"{key}: {value}" for key, value in self.items())
+        return f"{{{items}}}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.__calculate_index(key)
+
+        if self.__table[index].key is None:
+            if self.__size + 1 > self.__threshold:
+                self.__resize()
+                return self.__setitem__(key, value)
```

Comment:
You can refactor this so you don't have to use recursion here. You need to use recursion with caution as it may increase algorithmic complexity dramatically.
So here you can check if table needs resizing, resize if necessary, then calculate index and set node.